src/Service.php

```diff
@@ -7,179 +8,315 @@
         ];
+    }
 
-        $operationGroups = [];
-
-        // if there is a routeIdentifierName, this is REST service, need to enumerate
-        if ($service->routeIdentifierName) {
-            $entityOperations     = [];
-            $collectionOperations = [];
-
-            // find all COLLECTION operations
-            foreach ($service->operations as $collectionOperation) {
-                $method = $collectionOperation->getHttpMethod();
-
-                // collection parameters
-                $collectionParameters = $templateParameters;
-                unset($collectionParameters[$service->routeIdentifierName]);
-                $collectionParameters = array_values($collectionParameters);
-
-                if (in_array($method, ['POST', 'PUT', 'PATCH'])) {
-                    $collectionParameters[] = $postPatchPutBodyParameter;
-                }
-
-                $collectionOperations[] = [
-                    'method'           => $method,
-                    'summary'          => $collectionOperation->getDescription(),
-                    'notes'            => $collectionOperation->getDescription(),
-                    'nickname'         => $method . ' for ' . $service->getName(),
-                    'type'             => $service->getName(),
-                    'parameters'       => $collectionParameters,
-                    'responseMessages' => $collectionOperation->getResponseStatusCodes(),
-                ];
+    protected function isMethodPostPutOrPatch($method)
+    {
+        return in_array($method, ['post', 'put', 'patch']);
+    }
+
+    protected function getMethodFromOperation(Operation $operation)
+    {
+        return strtolower($operation->getHttpMethod());
+    }
+
+    protected function getPathOperation(Operation $operation, $parameters)
+    {
+        return $this->cleanEmptyValues([
+                'tags' => [$this->service->getName()],
+                'description' => $operation->getDescription(),
+                'parameters' => $parameters,
+                'produces' => $this->service->getRequestAcceptTypes(),
+                'responses' => $this->getResponsesFromOperation($operation)
+        ]);
+    }
+
+    protected function getResponsesFromOperation(Operation $operation)
+    {
+        $responses = [];
+        $responseStatusCodes = $operation->getResponseStatusCodes();
+        foreach ($responseStatusCodes as $responseStatusCode) {
+            $code = intval($responseStatusCode['code']);
+            $responses[$code] = $this->cleanEmptyValues([
+                'description' => $responseStatusCode['message'],
+                'schema' => $this->getResponseSchema($operation, $code)
+            ]);
+        }
+        return $responses;
+    }
+
+    protected function getResponseSchema(Operation $operation, $code)
+    {
+        if ($code === 200 || $code === 201) {
+            return $this->modelGenerator->generate($operation->getResponseDescription());
+        }
```

Comment:
Is it expected to have a `void`/`null` return type for non-200/201 response types here?